azext_iot/tests/dps/__init__.py

```diff
@@ -0,0 +1,258 @@
+            "version": None,
+        }
+        self.kwargs["generic_dict"] = {
+            **base_enrollment_props,
+            "key": "value",
+        }
+        self.kwargs["twin_array_dict"] = {
+            **base_enrollment_props,
+            "values": [{"key1": "value1"}, {"key2": "value2"}],
+        }
+
+        # Other variables for DPS testing
+        self.hub_host_name = "{}.azure-devices.net".format(ENTITY_HUB_NAME)
+
+    def create_dps(self):
+        """Create a device provisioning service for testing purposes."""
+        dps_list = self.cmd(
+            '''iot dps list -g "{}"'''.format(self.entity_rg)
+        ).get_output_in_json()
+
+        # Check if the generated name is already used
+        target_dps = None
+        for dps in dps_list:
+            if dps["name"] == self.entity_dps_name:
+                target_dps = dps
+                break
+
+        # Create the min version dps and assign the correct roles
+        if not target_dps:
+            self.cmd(
+                "iot dps create --name {} --resource-group {} ".format(
+                    self.entity_dps_name, self.entity_rg
+                )
+            )
+
+    def create_hub(self):
+        """Create an IoT hub for DPS testing purposes."""
+        hubs_list = self.cmd(
+            '''iot hub list -g "{}"'''.format(self.entity_rg)
+        ).get_output_in_json()
+
+        # Check if the generated name is already used
+        target_hub = None
+        for hub in hubs_list:
+            if hub["name"] == self.entity_hub_name:
+                target_hub = hub
+                break
+
+        # Create the min version hub and assign the correct roles
+        if not target_hub:
+            self.cmd(
+                "iot hub create --name {} --resource-group {} --sku S1 ".format(
+                    self.entity_hub_name, self.entity_rg
+                )
+            )
+
+    def _ensure_dps_hub_link(self):
+        hubs = self.cmd(
+            "iot dps linked-hub list --dps-name {} -g {}".format(self.entity_dps_name, self.entity_rg)
+        ).get_output_in_json()
+        if not len(hubs) or not len(
+            list(
+                filter(
+                    lambda linked_hub: linked_hub["name"]
+                    == "{}.azure-devices.net".format(self.entity_hub_name),
+                    hubs,
+                )
+            )
+        ):
+            self.cmd(
+                "iot dps linked-hub create --dps-name {} -g {} --connection-string {} --location {}".format(
+                    self.entity_dps_name, self.entity_rg, self.get_hub_cstring(), self.get_hub_region()
+                )
+            )
+
+    def _cleanup_enrollments(self):
+        """Delete all individual and group enrollments from the DPS."""
+        enrollments = self.cmd(
+            "iot dps enrollment list --dps-name {} -g  {}".format(self.entity_dps_name, self.entity_rg)
+        ).get_output_in_json()
+        if len(enrollments) > 0:
```

Comment:
nit - could do `if enrollments:` and evaluate list truthyness. Some more areas in this module could apply the same change.